### ISSUE TITLE
Dota 2 crashes on server connection attempt

### ISSUE THREAD

**Issue by Veroalla** [ash47/LegendsOfDota#52]
When I connect to any server through link steam://connect/lod.ash47.net:27015 my dota client crashes. 
However, using the console, it runs fine and showing version 2.3.2 LOD installed

This is happening to my friends as well running latest LOD. How can I troubleshoot this?

**Comment by ash47** [comment]
Is your client running 2.3.2?

**Comment by Veroalla** [comment]
Affirmative. So are my friends, we confirmed it by connecting through console. (which crashes on entry too, but less frequently)

**Comment by ash47** [comment]
How did you confirm it through the console?

**Comment by ash47** [comment]
Just download the latest release and install it, ensure you restart dota

**Comment by Veroalla** [comment]
Let me rephrase... I connected through the console and got to the screen "wait while we waste your time" and then a successful LOD version showed up with 2.3.2

**Comment by ash47** [comment]
When does it crash then?

**Comment by Veroalla** [comment]
It crashes a millisecond after the initial red Dota 2 loading screen

**Comment by ash47** [comment]
Has it crashed for you, or only your friends?

**Comment by Veroalla** [comment]
All of us, repeatedly. It's probably something Valve messed up.
We noticed it crashes less if we let Dota idle at main menu a bit, and then connect through console

**Comment by ash47** [comment]
I can't reproduce this, can you try removing lod/scripts/NPC_units_custom.text, and see if it still crashes? Just to confirm that is the issue, if so, I'll patch it some more

**Comment by Veroalla** [comment]
It still crashes 99% of the time after removing the file.
I noticed the game finishes loading faster when I launch dota from link. Does it have something to do with the crashes?
It usually loads 5+ secs longer if I run dota normally

**Comment by ash47** [comment]
This is really annoying because I can't reproduce, can you try deleting lod/resource/flash 3/custom ui.txt and reporting back to me?

**Comment by Veroalla** [comment]
Still the same issue. You mentioned flash 3, got me thinking about removing Lobby Explorer could fix it, but to no avail.

I tried to connect locally to my own server through link, same crash

**Comment by ash47** [comment]
Can you just delete the whole flash3 folder? The aim is to find out what is causing it

**Comment by Veroalla** [comment]
I deleted the whole flash3 folder (\addons\lod\resource) and in (\dota\resource), still crashing!
I can attach my server console if you could find some hints. I launched the server and tried to connect, then got dropped.
https://www.dropbox.com/s/rz456vgipb1dvs3/console.log?dl=1

**Comment by ash47** [comment]
Are you deleting on the client or server? Try deleting the scripts folder on the client?

**Comment by Veroalla** [comment]
Only from client.

My observations:
To fix this, you have to launch Dota 2 first. wait few secs and THEN you can click the connect link for your server. Success!

**Comment by ash47** [comment]
2.2.6 should work, if you want to play, sigh, I'll look into it more when I'm on a pc

**Comment by Veroalla** [comment]
i'll test rolling back my server and client. brb

**Comment by ash47** [comment]
Oh you're fucking joking, I knew that from d2ware days, kill me now

**Comment by ash47** [comment]
You do need to load Dota first, I will highlight this in the guide later

**Comment by Veroalla** [comment]
Holy Crap lol
I've had dozens of friends ragequitting when connecting from link and saying console is too complicated

**Comment by ash47** [comment]
Link is fine, just don't click until Dota is launched

**Comment by Veroalla** [comment]
So yeah. Thanks for your patience! >.<
It's not your LODs fault :DD yaaay

**Comment by ash47** [comment]
This makes me very happy as we were just dealing with a real crash earlier and I thought I had fixed it, hours and hours today working on it

**Comment by Veroalla** [comment]
I'm happy it had nothing to do with it then. Srry for worrying you :(
Anyway, I updated it here in the LOD auto-installer too: https://github.com/Veroalla/LOD-Auto-Installer

**Comment by ash47** [comment]
Cool, happy to help
